src/runtime_src/core/common/smi.h

```diff
@@ -148,6 +152,50 @@ class smi {
 
 };
 
+class config_generator {
+public:
+  virtual ~config_generator() = default;
+
+  XRT_CORE_COMMON_EXPORT
+  virtual subcommand 
+  create_validate_subcommand() = 0;
+
+  XRT_CORE_COMMON_EXPORT
+  virtual subcommand 
+  create_examine_subcommand() = 0;
+
+  XRT_CORE_COMMON_EXPORT
+  virtual subcommand 
+  create_configure_subcommand() = 0; 
+};
+
+/* This class includes all the hardware specific logic. 
+   Each shim should inherit from this if they are to define
+   shim specific hardware specific behavior*/
```

Comment:
Use C++ commenting '//'.  The comment does not make sense.   It it not this class that is supposed to be derived from ?